shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs

```diff
@@ -81,19 +84,69 @@ goldenTests =
         "addrEnterpriseS"
         putAddr
         (Addr scriptHash StakeRefNull)
-        "7205060708"
+        "7205060708",
+
+      -- serialisation golden tests
+      goldenSerialisation
+        "addrEnterpriseK for network id = 0"
+        (Addr (keyBlake2b224 paymentKey) StakeRefNull)
+        "62",
+      goldenSerialisation
+        "addrBaseKK for network id = 0"
+        (Addr (keyBlake2b224 paymentKey) (StakeRefBase (keyBlake2b224 stakeKey)))
+        "02",
+      goldenSerialisation
+        "addrPtrK for network id = 0"
+        (Addr (keyBlake2b224 paymentKey) (StakeRefPtr ptr))
+        "4281000203"
+
+       -- the same as above for other network ids is also needed, for example, for 3, 6
     ]
 
 golden :: String -> (a -> B.Put) -> a -> LBS.ByteString -> TestTree
 golden name put value expected =
   T.testCase name $
     T.assertEqual name expected (LB16.encode . B.runPut . put $ value)
 
+goldenSerialisation :: String -> C.Addr -> BS.ByteString -> TestTree
```

Comment:
it's a bit strange to use the "concrete crypto types" (ie `C`) here, since that is the mock crypto, which includes the short hash algorithm. If we want types with blake2b-244, we should use a different `crypto` type.

Reply:
yes!!! thanks -> https://github.com/input-output-hk/cardano-ledger-specs/pull/1443/commits/8c7a0d69f361f4195edd7ed89203813d7aac3d77